hpx/config.hpp

```diff
@@ -33,9 +33,9 @@
 
 #include <boost/version.hpp>
 
-#if BOOST_VERSION < 105800
+#if BOOST_VERSION < 106100
```

Comment:
Can we leave this unchanged? It will most likely work, we just don't support it anymore. Perhaps a #warning instead?

Reply:
Good point. I can do that.

Reply:
Hmm. Thinking about this a bit more, I'd rather not just warn about it this. The only versions that could work are 1.59 and 1.60, anything older definitely won't work. A warning can be too easily overlooked, so I'd rather just be explicit and strict about it. If anything maybe we don't need to bump the minimum version 1.61, but rather 1.59 or something, but sticking to the last 10 releases is much more straightforward.

Reply:
This should be a hard error, unless perhaps explicitly requested otherwise (will void warranty). Whether there's value in making this a supported option vs letting people try their luck and patch the code, I do not know.

Reply:
Ok, fair point.